### PR TITLE
New version + new architectures + switch to rattler-build

### PR DESCRIPTION
**Pull request by hadim**
Checklist
* [x] Used a [personal fork of the feedstock to propose changes](https://conda-forge.org/docs/maintainer/updating_pkgs.html#forking-and-pull-requests)
* [x] Bumped the build number (if the version is unchanged)
* [x] Reset the build number to `0` (if the version changed)
* [x] [Re-rendered]( https://conda-forge.org/docs/maintainer/updating_pkgs.html#rerendering-feedstocks ) with the latest `conda-smithy` (Use the phrase <code>@<space/>conda-forge-admin, please rerender</code> in a comment in this PR for automated rerendering)
* [x] Ensured the license file is being packaged.

### CODE REVIEW

**Comment by kylebgorman** [comment]
Lots of changes here but I don't know what "rattler-build" is: is there a doc that gives some context?

**Comment by hadim** [comment]
> Lots of changes here but I don't know what "rattler-build" is: is there a doc that gives some context?

Wow you're fast! I will give more context in my next message.

rattler-build is an alternative to conda-build. More infos at https://prefix-dev.github.io/rattler-build/latest/ and https://prefix.dev/blog/rattler_build_on_conda_forge

**Comment by hadim** [comment]
I am trying to build a package called `deepsearch-glm` at https://github.com/conda-forge/staged-recipes/pull/28093 and it requires `fasttext`. Unfortunately, `deepsearch-glm` needs a forked and modified version of the package, and it's worth noting the original repo (from facebookresearch) is also archived and unmaintained.

The needed fork does not have any tag, and it might be nice to have one to build a package from it here.

@PeterStaar-IBM would that be possible for you to push a git tag at https://github.com/PeterStaar-IBM/fastText? Something like 0.10.0 will do the job.

@conda-forge/fasttext would you be okay, building a new version here from @PeterStaar-IBM's fork?

---

To summarize what this PR contains:

- enable osx_arm64 and linux_aarch64/linux_ppc64le
	- replace https://github.com/conda-forge/fasttext-feedstock/pull/21
- Build using latest from  https://github.com/PeterStaar-IBM/fastText
- Switch build engine to `rattler-build`

**Comment by PeterStaar-IBM** [comment]
> I am trying to build a package called `deepsearch-glm` at [conda-forge/staged-recipes#28093](https://github.com/conda-forge/staged-recipes/pull/28093) and it requires `fasttext`. Unfortunately, `deepsearch-glm` needs a forked and modified version of the package, and it's worth noting the original repo (from facebookresearch) is also archived and unmaintained.
> 
> The needed fork does not have any tag, and it might be nice to have one to build a package from it here.
> 
> @PeterStaar-IBM would that be possible for you to push a git tag at https://github.com/PeterStaar-IBM/fastText? Something like 0.10.0 will do the job.
> 
> @conda-forge/fasttext would you be okay, building a new version here from @PeterStaar-IBM's fork?
> 
> To summarize what this PR contains:
> 
> * enable osx_arm64 and linux_aarch64/linux_ppc64le
>   
>   * replace [ARM OSX Migrator #21](https://github.com/conda-forge/fasttext-feedstock/pull/21)
> * Build using latest from  https://github.com/PeterStaar-IBM/fastText
> * Switch build engine to `rattler-build`

Hi, @hadim , yes, just did that here: https://github.com/PeterStaar-IBM/fastText/releases/tag/v0.1.0

**Comment by kylebgorman** [comment]
I am not a "core maintainer" of anything but the conda-forge repo but I think replacing base fasttext with a fork (and not changing the name) might be controversial. Let me suggest you run it by some people on the Conda-Forge Gitter system first (https://conda-forge.org/community/getting-in-touch/) to see what maintainers think in general.

**Comment by hadim** [comment]
> > I am trying to build a package called `deepsearch-glm` at [conda-forge/staged-recipes#28093](https://github.com/conda-forge/staged-recipes/pull/28093) and it requires `fasttext`. Unfortunately, `deepsearch-glm` needs a forked and modified version of the package, and it's worth noting the original repo (from facebookresearch) is also archived and unmaintained.
> > The needed fork does not have any tag, and it might be nice to have one to build a package from it here.
> > @PeterStaar-IBM would that be possible for you to push a git tag at https://github.com/PeterStaar-IBM/fastText? Something like 0.10.0 will do the job.
> > @conda-forge/fasttext would you be okay, building a new version here from @PeterStaar-IBM's fork?
> > To summarize what this PR contains:
> > 
> > * enable osx_arm64 and linux_aarch64/linux_ppc64le
> >   
> >   * replace [ARM OSX Migrator #21](https://github.com/conda-forge/fasttext-feedstock/pull/21)
> > * Build using latest from  https://github.com/PeterStaar-IBM/fastText
> > * Switch build engine to `rattler-build`
> 
> Hi, @hadim , yes, just did that here: https://github.com/PeterStaar-IBM/fastText/releases/tag/v0.1.0

Thank you @PeterStaar-IBM but it's 0.1.0. Do you mind pushing **0.10.0** instead?

**Comment by hadim** [comment]
> I am not a "core maintainer" of anything but the conda-forge repo but I think replacing base fasttext with a fork (and not changing the name) might be controversial. Let me suggest you run it by some people on the Conda-Forge Gitter system first (https://conda-forge.org/community/getting-in-touch/) to see what maintainers think in general.

Yes, I agree with you, it's not ideal, but on the other side the upstream repo is also archived on GH, so there is no chance for a new version.

@conda-forge/core do you think it's ok to proceed here, or is that best if create an entirely new package and submit it on staged-recipes instead? I am happy with both.

**Review by beckermr — CHANGES_REQUESTED** [review]
Yes we cannot point feedstocks at forks like this. 

I am not sure what to do though.

**Comment by hadim** [comment]
> Yes we cannot point feedstocks at forks like this.
> 
> I am not sure what to do though.

Ok so what about creating a new package on staged-recipes and call it something like `fasttext-fork` or `fasttext-for-deepsearch-glm`?

**Comment by hadim** [comment]
Alternative PR at https://github.com/conda-forge/staged-recipes/pull/28098

**Comment by hadim** [comment]
Closing here in favor of https://github.com/conda-forge/staged-recipes/pull/28098

Thanks everyone, for the inputs.
